### PR TITLE
[Merged by Bors] - refactor: purge aesop_cat_nonterminal

### PR DESCRIPTION
**Pull request by JLimperg**
`aesop_cat_nonterminal` is a non-terminal variant of `aesop`. It's not supposed to be used in production code since it's even worse than non-terminal `simp`. However, there were a few occurrences left (presumably from the port), which this PR removes.

The only nontrivial change is that I add mathlib's `rfl` tactic to the `CategoryTheory` Aesop rule set.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Category/ModuleCat/Algebra.lean [hunk 1]
```diff
@@ -65,9 +65,7 @@ instance linearOverField : Linear k (ModuleCat.{v} A) where
   homModule M N := LinearMap.module
   smul_comp := by
     -- Porting note: this was automatic by `aesop_cat`
-    aesop_cat_nonterminal
-    rw [LinearMap.smul_apply, LinearMap.smul_apply, LinearMap.map_smul_of_tower]
-    rfl
+    intros; ext; erw [LinearMap.map_smul_of_tower]; rfl
```
Comment (semorrison) [inline Mathlib/Algebra/Category/ModuleCat/Algebra.lean:68]: Why was `erw` not necessary after `aesop_cat_nonterminal` but is needed now? I would prefer a `dsimp; rw` over `erw` any day: `erw` is a sign something is wrong.

File: Mathlib/CategoryTheory/Sums/Basic.lean [hunk 2]
```diff
@@ -136,15 +144,13 @@ theorem swap_map_inr {X Y : D} {f : inr X ⟶ inr Y} : (swap C D).map f = f :=
...
 namespace Swap
 
-/- Porting note: had to manually call `cases f` for `f : PEmpty` -/
-
 /-- `swap` gives an equivalence between `C ⊕ D` and `D ⊕ C`. -/
 def equivalence : Sum C D ≌ Sum D C :=
   Equivalence.mk (swap C D) (swap D C)
     (NatIso.ofComponents (fun X => eqToIso (by cases X <;> rfl))
-      (by simp only [swap]; aesop_cat_nonterminal; cases f; cases f))
+      (by aesop_cat (add norm simp swap)))
```
Comment (semorrison) [inline Mathlib/CategoryTheory/Sums/Basic.lean:151]: You can just drop this argument. `aesop_cat` can do it unaided, and it is the autoparam.

File: Mathlib/CategoryTheory/Sums/Basic.lean [hunk 3]
```diff
@@ -136,15 +144,13 @@ theorem swap_map_inr {X Y : D} {f : inr X ⟶ inr Y} : (swap C D).map f = f :=
...
     (NatIso.ofComponents (fun X => eqToIso (by cases X <;> rfl))
-      (by simp only [swap]; aesop_cat_nonterminal; cases f; cases f))
+      (by aesop_cat (add norm simp swap)))
```
Comment (semorrison) [inline Mathlib/CategoryTheory/Sums/Basic.lean:153]: And here.

**Comment by semorrison** [comment]
Otherwise, LGTM.

bors d+

File: Mathlib/CategoryTheory/Sums/Basic.lean (hunk 2 above)
Comment (JLimperg) [reply Mathlib/CategoryTheory/Sums/Basic.lean]: Done x2.

File: Mathlib/Algebra/Category/ModuleCat/Algebra.lean (hunk 1 above)
Comment (JLimperg) [reply Mathlib/Algebra/Category/ModuleCat/Algebra.lean]: I managed to avoid the `erw` with a combination of `dsimp` and `rw`.

**Comment by JLimperg** [comment]
Thanks for the review!

bors r+

**Pull request closed without merging**